server/modules/taxonomy/repository/taxonomyRepository.js

```diff
@@ -28,44 +33,35 @@ export const insertTaxonomy = async (surveyId, taxonomy, client = db) =>
     record => dbTransformCallback(record, true, true)
   )
 
-export const insertTaxa = async (surveyId, taxa, client = db) =>
-  await client.batch(R.reduce((acc, taxon) => {
-      const taxonInsertPromise = insertOrUpdateTaxon(surveyId, taxon, client)
-
-      const vernacularNameInsertPromises = insertOrUpdateVernacularNames(
-        surveyId, Taxon.getUuid(taxon), Taxon.getVernacularNames(taxon), client)
-
-      return R.pipe(
-        R.append(taxonInsertPromise),
-        R.concat(vernacularNameInsertPromises)
-      )(acc)
-    }, [], taxa)
+const _insertOrUpdateVernacularNames = (surveyId, taxonUuid, vernacularNames, client = db) =>
+  Object.entries(vernacularNames).map(([lang, vernacularName]) =>
+    client.none(
+      `INSERT INTO 
+           ${getSurveyDBSchema(surveyId)}.taxon_vernacular_name (uuid, taxon_uuid, props_draft)
+        VALUES 
+          ($1, $2, $3)
+        ON CONFLICT 
+          (taxon_uuid, ((props||props_draft)->>'${TaxonVernacularName.keysProps.lang}')) DO
+        UPDATE SET 
+          props_draft = ${getSurveyDBSchema(surveyId)}.taxon_vernacular_name.props_draft || $3`,
+      [TaxonVernacularName.getUuid(vernacularName), taxonUuid, TaxonVernacularName.getProps(vernacularName)]
+    )
```

Comment:
what happens if you have same lang for more than one vernacular name?

Reply:
only one vernacular name with that language will be stored (this will be addressed in another task)